### ISSUE TITLE
As an Open Data Hub Manager I want to have a Code Development guideline for the project, to help new developers to contribute to the project.

### ISSUE THREAD

**Issue by sseppi** [noi-techpark/it.bz.opendatahub.databrowser#299]
The tasks of this issue is:
- [x] Code analysis
- [x] Presentation to the project team of the analysis results that includes
  - Proposals of optimization
  - List of pro/contra of the possible solutions
  - Solution suggestions
- [x] The team decides how to g o on in case of doubts
- [x] The guideline is added in the repo as Guideline.md file
- [x] The repo will be refactored (a deidcated issue has been created - #337 )

**Comment by sseppi** [comment]
@MatteoBiasi 
We had a meeting yesterday where we decided to take some default configuration (e.g. read only data for the Edit View) directly from the Open API. Considering that we decided also that, once we will have this feature, the Web Component should be implemented in the way that if the Open API doesn't provide any information the data will be read only per default.

We should consider to add this in the Development Guideline.
